src/semidiscretization/semidiscretization_hyperbolic_parabolic.jl

```diff
@@ -330,4 +330,25 @@ function rhs_parabolic!(du_ode, u_ode, semi::SemidiscretizationHyperbolicParabol
 
     return nothing
 end
+
+function rhs_hyperbolic_parabolic!(du_ode, u_ode,
+                                   semi::SemidiscretizationHyperbolicParabolic, t)
+    @trixi_timeit timer() "hyperbolic-parabolic rhs!" begin
+        # Implementation of split ODE problem in OrdinaryDiffEq
+        du_ode_hyp = similar(du_ode)
+        rhs!(du_ode_hyp, u_ode, semi, t)
+        rhs_parabolic!(du_ode, u_ode, semi, t)
+        du_ode .+= du_ode_hyp
+    end
+end
+
+function _jacobian_ad_forward(semi::SemidiscretizationHyperbolicParabolic, t0, u0_ode,
+                              du_ode, config)
+    new_semi = remake(semi, uEltype = eltype(config))
+    J = ForwardDiff.jacobian(du_ode, u0_ode, config) do du_ode, u_ode
+        Trixi.rhs_hyperbolic_parabolic!(du_ode, u_ode, new_semi, t0)
+    end
+
+    return J
+end
```

Comment:
I don't really like that this will allocate a temporary vector for each call of `rhs_hyperbolic_parabolic!`. Can you please pre-allocate it? 
Moreover, I am not sure whether having `rhs_hyperbolic_parabolic!` will be confusing since people will expect it to be used by hyperbolic parabolic semidiscretizations.

Reply:
Agreed with both points.
The allocation could e.g. be done in 

```julia

function _jacobian_ad_forward(semi::SemidiscretizationHyperbolicParabolic, t0, u0_ode,
                              du_ode, config)
```

The name of the function is somewhat misleading - one should add comments regarding this.

Reply:
You don't need a new function for it but could just define it in the `do` block, couldn't you?

Reply:
True - I had `rhs_hyperbolic_parabolic` still lying around due to the issue in OrdinaryDiffEq.

Reply:
Addressed in [490d29b](https://github.com/trixi-framework/Trixi.jl/pull/1589/commits/490d29bf16bf42b9044170885e35e84f7733f908)